### ISSUE TITLE
Convert to wyam.io

### ISSUE THREAD

**Issue by SeanKilleen** [AWiderDotNET/AWiderDotNet.github.io#3]
I love Jekyll and it made the MVP easy to set up very quickly with Github pages.

However, I'd love to make use of [Wyam.io](http://wyam.io) for this project, to showcase yet another great .NET OSS Project.

## Remaining Work
* [x] Get wyam.io brought in and installed (#28)
* [x] Ensure wyam builds the site (#28)
* [x] Ensure wyam compiles the SASS and looks the same. (#28)
* [ ] Come up with the scripted build for wyam.io  (#28)
* [ ] Make any modifications to `.travis.yml` necessary to get the branch pushes right
* [ ] Doublecheck the `gh-pages` branch output
* [ ] Switch the github pages settings to point to the `gh-pages` branch.

**Comment by agc93** [comment]
I know @daveaglick and he was a massive help in getting us switched over to Wyam when our project moved not long ago. He might be able to help out 😄

**Comment by SeanKilleen** [comment]
For sure, he'd be the guy I go to. :) It'd be a great opportunity to work with Wyam, which I've been looking to do for a while now.

**Comment by daveaglick** [comment]
![img_4230](https://cloud.githubusercontent.com/assets/1020407/26793844/5c7c945e-49d4-11e7-9d28-e28652507664.GIF)

Someone rang...

Would love to help out, and I'm here! Let's all get together (maybe lunch?) and see what we can do.

**Comment by khalidabuhakmeh** [comment]
Wyam is cool, but I feel like the move to wyam may inadvertently cast a smaller net of contributions to your goal. Since you've chosen GitHub, it may be in the projects interest to keep it on Jekyll. Less infrastructure is needed and contributions to the site can happen from cross platform users.

Anways, my 2 cents.

**Comment by daveaglick** [comment]
I'm obviously a little biased, but to be honest I find Jekyll much harder to set up for a local build and test environment (at least on Windows). That's especially true for publishing to GitHub since it uses some special Jekyll extensions. Obviously non-Windows local building is easier with Jekyll since that's not even an option with Wyam yet, though hopefully that'll be addressed relatively soon.

The integrated Jekyll builds in GitHub are certainly a consideration for the maintainer and setting up a CI/CD pipeline, but I don't think that extends to contributors. It's not like when you submit a PR to a Jekyll-based GitHub Pages site that each PR comes with a hosted preview of the changes. That all happens out of band from GitHub and gets back to locally building and testing. The Jekyll integration is only relevant for the finished product, which the broader community rarely interacts directly with.

There's also the issue of how far this site is intended to go in terms of complex build logic. Jekyll is great for templated builds that color inside the lines, but gets tough quickly for more complex scenarios, especially if Ruby isn't your thing. On the other hand, Wyam is easier to customize and extend using C# (though my bias might be showing here again).

All that said, I'm pragmatic more than anything else. I've never pushed Wyam where a maintainer or community doesn't think it's appropriate and don't intend to. If it turns out it's the wrong approach here then no hard feelings. If not, some issues on the Wyam repo with suggested improvements so it could be more helpful in the future would be appreciated.

**Comment by khalidabuhakmeh** [comment]
@daveglick you make some good points. I don't think anything is wrong with Wyam and I hope it didn't come across as doubting the project itself. I was merely saying that Jekyll is a first class citizen of the GitHub ecosystem and may keep things simpler.

That said, the issues still outstanding in this issue would go a long way to helping others contribute to the site. If the common workflow is to edit an existing markdown file or add a new markdown file, then everyone wins :)

Curious to see what a designer thinks about the topic.

:hugs:

**Comment by daveaglick** [comment]
@khalidabuhakmeh No offense taken - if my reply seemed defensive, it's only because I'm generally long-winded. This isn't the first time the "Jekyll is native to GitHub" point has been made in static projects and it's defintly one that should at least be considered.

As for design - I've suggested to @SeanKilleen that we look at themes from https://html5up.net once this PR is merged (or not). I was planning on a follow-up PR with a new design, but would gladly yield to anyone else who wants to take up that aspect.

**Comment by khalidabuhakmeh** [comment]
That site is pretty cool.

It's unclear what @seankilleen wants on the site and what he wants the focal point to be. Here is what I can glean from the issues and the existing site.

- [ ] blog
- [ ] Community resource/ wiki
- [ ] Video archive
- [ ] Discussions 

Also most likely the mission statement of the project clearly defined.

I understand the "what" part, the "how" part is still kind of fuzzy to me. Elaborating the "how" would be helpful.

**Comment by khalidabuhakmeh** [comment]
Ok I'm sitting at the airport so I have time to think. 
Reading the content of the site I feel like existing content is a perfect opportunity to gameify what you wa

**Comment by khalidabuhakmeh** [comment]
Fat finger...

Gameify what you want. The levels could be kinds of fish you may catch with a net. Sardines all the way up to blue whale.

These "rewards" or "badges" could be earned by companies that exhibit the behaviors you are promoting.

That would make the site an experience.

**Comment by SeanKilleen** [comment]
Hey all! Sorry, I'd been blissfully away from tech for a bit so just catching up on this.

@khalidabuhakmeh, for vision for this project, check my .NET Fringe talk video. The goal is for this site to become a place to help move companies and developers to the next rung on the OSS contribution maturity model as I defined it, and to refine the model itself. The site in its current form is a bare bones representation of that. for more, see the other repo in the AWiderDotNet org which contains some previous discussions.

Regarding wyam vs Jekyll (though I understand that's not the way anyone's framing it), I personally am more familiar with Jekyll and recognize it's ease of integration with GH pages. However, I see an opportunity to make use of a great .NET based OSS project that's had good care taken in its design, and my intention is for the content to remain largely markdown-driven. If folks have problems contributing, I'm happy to reconsider as it comes to that, but with a solid build process and the awesome contribution of @daveaglick so far, I think we can give it a go.
